### ISSUE TITLE
[tune] Error handling checkpoint when using checkpoint_score_attr and keep_checkpoints_num

### ISSUE THREAD

**Issue by Lawiss** [ray-project/ray#7083]
Version info :
- Ray 0.8.1
- Python 3.7.3

Hi,
I'm using tune with the Trainable interface. 
I'm trying to keep only the best checkpoint of each trial, thus this I've configured tune.run to do so :
```
analysis = tune.run(
        trainable,
        scheduler=sched,
        stop={
            "r2_mean": 0.5,
            "training_iteration": 3 if args.smoke_test else 30,
        },
        resources_per_trial={
            "cpu": 6,
            "gpu": int(args.use_gpu)
        },
        num_samples=1 if args.smoke_test else 70,
        checkpoint_at_end=True,
        checkpoint_freq=2,
        checkpoint_score_attr="r2_mean",
        keep_checkpoints_num=1,
        config=search_space,
        local_dir="./ray_results_roberta")
```
The problem appears when the second checkpoint of a trial is about to be saved and the checkpoint manager attempt to remove the old checkpoint, I have the following error : 
```
2020-02-07 09:07:00,011 ERROR ray_trial_executor.py:585 -- Trial FullTrainable_9f6d5e76: Error handling checkpoint /home/stageIA/datadrive/Projects/Kaggle/Google_qa/ray_results_roberta/FullTrainable/FullTrainable_9f6d5e76_2020-02-07_08-34-14q43cnxqu/checkpoint_3/model.pt
h                                                                                                                                                        
Traceback (most recent call last):                                                                                                                       
  File "/home/stageIA/datadrive/anaconda3/envs/transformers/lib/python3.7/site-packages/ray/tune/ray_trial_executor.py", line 582, in save               
    trial.on_checkpoint(checkpoint)                                     
  File "/home/stageIA/datadrive/anaconda3/envs/transformers/lib/python3.7/site-packages/ray/tune/trial.py", line 399, in on_checkpoint
    self.checkpoint_manager.on_checkpoint(checkpoint)                   
  File "/home/stageIA/datadrive/anaconda3/envs/transformers/lib/python3.7/site-packages/ray/tune/checkpoint_manager.py", line 114, in on_checkpoint
    self.delete(worst)                                                  
  File "/home/stageIA/datadrive/anaconda3/envs/transformers/lib/python3.7/site-packages/ray/tune/trial.py", line 91, in delete
    shutil.rmtree(checkpoint_path)                                                             
  File "/home/stageIA/datadrive/anaconda3/envs/transformers/lib/python3.7/shutil.py", line 491, in rmtree
    _rmtree_safe_fd(fd, path, onerror)                                  
  File "/home/stageIA/datadrive/anaconda3/envs/transformers/lib/python3.7/shutil.py", line 410, in _rmtree_safe_fd
    onerror(os.scandir, path, sys.exc_info())                    
  File "/home/stageIA/datadrive/anaconda3/envs/transformers/lib/python3.7/shutil.py", line 406, in _rmtree_safe_fd
    with os.scandir(topfd) as scandir_it:                        
NotADirectoryError: [Errno 20] Not a directory: '/home/stageIA/datadrive/Projects/Kaggle/Google_qa/ray_results_roberta/FullTrainable/FullTrainable_9f6d5e76_2020-02-07_08-34-14q43cnxqu/checkpoint_2/model.pth'
```
Indeed, the path points to a file and not a directory. 
If i do'nt use checkpoint_score_attr and keep_checkpoints_num (only checkpoint_freq), there is no problem and all checkpoints are correctly saved.
Is this a bug or did I miss something ?

**Comment by ujvl** [comment]
Yeah this was fixed by #6758 so I would recommend just installing the latest wheels.

**Comment by Lawiss** [comment]
Thanks, i'll try it!
